en/services/parameter_ext.md

```diff
@@ -0,0 +1,238 @@
+-->
+
+
+## Parameter Caching {#parameter_caching}
+
+A GCS or other component may choose to maintain a cache of parameter values for connected components/systems, in order to reduce the time required to display values and reduce MAVLink traffic.
+
+The cache can be populated initially by first [reading the full parameter list](#read_all) at least once, and then updated by monitoring for [PARAM_EXT_ACK](#PARAM_EXT_ACK) messages with `PARAM_ACK_ACCEPTED` (which are emitted whenever a parameter is successfully [written/changed](#write)).
+
+A system may also monitor for [PARAM_EXT_VALUE](#PARAM_EXT_VALUE) originating from other components/systems requesting parameter values. 
+
+> **Note** Cache synchronisation is not guaranteed; a component may [miss parameter update messages](#monitoring_unreliable) due to changes by other components.
+
+
+<!--
+## Multi-System and Multi-Component Support
+
+MAVLink supports multiple systems in parallel on the same link, and multiple MAVLink enabled components within a system.
+
+Requests to get and set parameters can be sent to individual systems or components. 
+To get a complete parameter list from a system, send the request parameter message with `target_component` set to [MAV_COMP_ID_ALL](../messages/common.md#MAV_COMP_ID_ALL).
+
+All components must respond to parameter request messages addressed to their ID or the ID `MAV_COMP_ID_ALL`. 
+
+> **Tip** *QGroundControl* by default queries all components of the currently connected system (it sends ID `MAV_COMP_ID_ALL`).
+-->
+
+
+<!--
+### Mavgen C API
+
+The C API provides a convenient `union` that allows you to bytewise convert between any of the supported types: `mavlink_param_union_t` ([mavlink_types.h](https://github.com/mavlink/c_library_v2/blob/master/mavlink_types.h)).
+For example, below we shown how you can set the union integer field but pass the float value to the sending function: 
+
+```c
+mavlink_param_union_t param;
+int32_t integer = 20000;
+param.param_int32 = integer;
+param.type = MAV_PARAM_TYPE_INT32;
+
+// Then send the param by providing the float bytes to the send function
+mavlink_msg_param_set_send(xxx, xxx, param.param_float, param.type, xxx);
+```
+
+### Mavgen Python API (Pymavlink)
+
+Pymavlink does not include special support to byte-wise encode the non-float data types (unsurprisingly, because Python effectively "hides" low level data types from users).
+When working with a system that supports non-float parameters (e.g. PX4) you will need to do the encoding/decoding yourself when sending and receiving messages.
+
+There is a good example of how to do this in the Pymavlink [mavparm.py](https://github.com/ArduPilot/pymavlink/blob/master/mavparm.py) module (see `MAVParmDict.mavset()`).
+
+-->
+
+## Limitations {#limitations}
```

Comment:
Note, please read these limitations. I believe they are the same as for the original parameters protocol